src/transformers/generation/logits_process.py

```diff
@@ -480,8 +480,8 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         cumulative_probs = sorted_logits.softmax(dim=-1).cumsum(dim=-1)
 
         # Remove tokens with cumulative mass above the threshold
-        last_ind = (cumulative_probs < self.mass).sum(dim=1)
-        last_ind[last_ind < 0] = 0
+        last_ind = (cumulative_probs < self.mass).sum(dim=1) - 1
+        last_ind.clamp_(0)
```

Comment:
```suggestion
        last_ind.clamp_(min=0)
```
nit: let's make it more clear that it is a minimum clamp

Reply:
@Rocketknight1 After the change, the `(cumulative_probs < self.mass).sum(dim=1)` can be 0, which added to `-1` turns into a negative value. The cumsum vector at index N already contains the value of the tensor at index N, so if the first member of `cumulative_probs` is larger than `self.mass` a negative value will come out of here :D 

Before the change, it seemed to be a redundant op, yes 😂

Reply:
That makes sense, lol. Just making sure I wasn't missing something incredibly obvious!